lib/puppet/util/monkey_patches.rb

```diff
@@ -30,6 +30,20 @@ def daemonize
   end
 end
 
+class Dir
+  def self.exists?(file_name)
+    Puppet.warning('exists? is a deprecated name, use exist? instead')
+    Dir.exist?(file_name)
+  end
```

Comment:
We may want to make this conditional so we only define it if it doesn't exist (when running puppet as a gem on Ruby 3.0 or 3.1). You can do something like what we do for OpenSSL lower down in the monkey patches (but have to check if the singleton (class) method is defined instead of instance method. Something like:

```ruby
unless File.singleton_methods.include?(:exists?)
  # monkey patch it
end
```